examples/taptree_of_horror/taptree_of_horror.rs

```diff
@@ -0,0 +1,314 @@
+    let a_descriptor_desc_secret = DescriptorSecretKey::from_str(&alice).unwrap();
+    let b_descriptor_desc_secret = DescriptorSecretKey::from_str(&bob).unwrap();
+    let c_descriptor_desc_secret = DescriptorSecretKey::from_str(&charlie).unwrap();
+    let d_descriptor_desc_secret = DescriptorSecretKey::from_str(&dave).unwrap();
+    let e_descriptor_desc_secret = DescriptorSecretKey::from_str(&eve).unwrap();
+    let f_descriptor_desc_secret = DescriptorSecretKey::from_str(&frank).unwrap();
+    let h_descriptor_desc_secret = DescriptorSecretKey::from_str(&heather).unwrap();
+    let i_descriptor_desc_secret = DescriptorSecretKey::from_str(&ian).unwrap();
+    let j_descriptor_desc_secret = DescriptorSecretKey::from_str(&judy).unwrap();
+    let l_descriptor_desc_secret = DescriptorSecretKey::from_str(&liam).unwrap();
+    let s_descriptor_desc_secret = DescriptorSecretKey::from_str(&s_backup_1).unwrap();
+    let x_descriptor_desc_secret = DescriptorSecretKey::from_str(&x_backup_2).unwrap();
+    let grim = produce_grim_hash("sovereignty through knowledge");
+    let kelly = produce_kelly_hash("the ultimate pre-preimage");
+
+    // define keys that will be used for the policy and for signing.
+    let internal_xpub: miniscript::DescriptorPublicKey =
+        internal_desc_secret.to_public(secp).unwrap();
+    let _intneral_xpriv: DescriptorXKey<Xpriv> = match internal_desc_secret.clone() {
+        DescriptorSecretKey::XPrv(x) => Some(x),
+        _ => None,
+    }
+    .unwrap();
+    let (a_pks, a_prvs) = produce_key_pairs(a_descriptor_desc_secret, secp, normal_path, "alice");
+    let (b_pks, b_prvs) = produce_key_pairs(b_descriptor_desc_secret, secp, normal_path, "bob");
+    let (c_pks, c_prvs) = produce_key_pairs(c_descriptor_desc_secret, secp, normal_path, "charlie");
+    let (d_pks, d_prvs) = produce_key_pairs(d_descriptor_desc_secret, secp, weird_path, "dave");
+    let (e_pks, e_prvs) = produce_key_pairs(e_descriptor_desc_secret, secp, normal_path, "eve");
+    let (f_pks, f_prvs) = produce_key_pairs(f_descriptor_desc_secret, secp, normal_path, "frank");
+    let (h_pks, h_prvs) = produce_key_pairs(h_descriptor_desc_secret, secp, normal_path, "heather");
+    let (i_pks, i_prvs) = produce_key_pairs(i_descriptor_desc_secret, secp, unhardened_path, "ian");
+    let (j_pks, j_prvs) = produce_key_pairs(j_descriptor_desc_secret, secp, normal_path, "judy");
+    let (l_pks, l_prvs) = produce_key_pairs(l_descriptor_desc_secret, secp, normal_path, "liam");
+    let (s_pks, _s_prvs) =
+        produce_key_pairs(s_descriptor_desc_secret, secp, normal_path, "s_backup1");
+    let (x_pks, _x_prvs) =
+        produce_key_pairs(x_descriptor_desc_secret, secp, normal_path, "x_backup2");
+
+    // define individual variables to make it easy to insert into the policy string.
+    // Define individual variables with pattern matching in a compact format
+    let (a0, a1, a2, a3, a4, a5, a6, a7, a8) = match &a_pks[..] {
```

Comment:
In  1282d30653a3cb4e188e59d5a3dd8de909aa9477

Can replace all these with one-liners, e.g.

```rust
let [a0, a1, a2, a3, a4, a5, a6, a7, a8] = a_pks[..].try_into().unwrap();
```

I would also drop the `Define individual variables with pattern matching` comment since it's just a repetition of the code.

Reply:
fyi, I'm also be including the explicit type to make the compiler/linter happy
`[PublicKey; KEYS_PER_PERSONA]`

Reply:
If you think that's necessary -- though I tried it without type annotations and it seemed to work. (Maybe you need the annotations once you convert all of them; I only did one.)

Reply:
I tried it again and removed the type for the "alice"/"a" keys and got this:

```
cargo +1.63.0 run --example taptree_of_horror --features compiler
   Compiling miniscript v12.2.0 (/Users/tzez/git/github/rust-bitcoin/rust-miniscript)
error[E0282]: type annotations needed
  --> examples/taptree_of_horror/taptree_of_horror.rs:88:9
   |
88 |     let [a0, a1, a2, a3, a4, a5, a6, a7, a8] = a_pks[..].try_into().unwrap();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: type must be known at this point
help: consider giving this pattern a type
   |
88 |     let [a0, a1, a2, a3, a4, a5, a6, a7, a8]: _ = a_pks[..].try_into().unwrap();
   |                                             +++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `miniscript` due to previous error
```

having the specific type solves this error, if you want to address it in a different way let me know.